libs/platform/core/src/services/storage/indexed-db-storage.ts

```diff
@@ -0,0 +1,56 @@
+import { Dexie, liveQuery } from 'dexie';
+import { Observable, shareReplay, switchMap } from 'rxjs';
+import {
+  indexedDbStorageName,
+  indexedDbTableName,
+  IndexedStorage,
+  StoredValue,
+} from './model';
+
+export class IndexedDbStorage implements IndexedStorage {
+  protected db = new Observable<Dexie>((subscriber) => {
+    const db = new Dexie(indexedDbStorageName);
+    db.version(1).stores({
+      [indexedDbTableName]: '&key,value',
+    });
+
+    subscriber.next(db);
+
+    return () => {
+      db.close();
+    };
+  }).pipe(
+    switchMap(async (db) => {
+      await db.open();
+
+      return db;
```

Comment:
You can return the table here so you don't have to do it every time in other places:
```suggestion
      return db.table(indexedDbTableName);
```